MFEM: add new version v4.8

v-dobrev opened the pull request:
TODO:
- [x] After MFEM v4.8 is released, update the version number (`4.8.0-dev` -> `4.8.0`) and its source
- [ ] Migrate to https://github.com/spack/spack-packages, e.g. using this tool: https://github.com/spack/migrate-package-prs
- [ ] Add `enzyme` variant

OPTIONAL:
- [ ] Add `magma` variant

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

v-dobrev committed:
[WIP] Add mfem v4.8: initial testing

v-dobrev commented:
@spackbot fix style

v-dobrev committed:
[@spackbot] updating style on behalf of v-dobrev

alecbcs commented on var/spack/repos/builtin/packages/mfem/package.py:
```diff
@@ -282,7 +285,7 @@ class Mfem(Package, CudaPackage, ROCmPackage):
 
     depends_on("mpi", when="+mpi")
     depends_on("hipsparse", when="@4.4.0:+rocm")
-    depends_on("hipblas", when="@4.4.0:+rocm")
+    depends_on("hipblas", when="@4.8.0:+rocm")
```
Why are we moving the lowerbound here? Does mfem < 4.8.0 not support hipblas?

v-dobrev replied:
Yes, before v4.8, MFEM does not depend (directly) on `hipblas`.

alecbcs replied:
Gotcha, so this was an error in the Spack package prior?

v-dobrev replied:
Yes.

v-dobrev committed:
[mfem] Set official mfem v4.8 source

v-dobrev commented:
Gitlab CI failures seem to be due to compiler crashes with GCC 7.x.x, e.g. here https://gitlab.spack.io/spack/spack/-/jobs/16371118/viewer#L35766.

v-dobrev commented:
We test with GCC 8 and we don't see compiler crashes with this version.

v-dobrev committed:
[mfem] Updates for v4.8

[magma] Add new versions for the rocm-core dependency

[spectrum-mpi] Do not query languages that are not used by a dependent package

[sundials] Fix llvm-amdgpu's C compiler: clang -> amdclang

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.8

v-dobrev commented:
@spackbot fix style

v-dobrev committed:
[@spackbot] updating style on behalf of v-dobrev

v-dobrev committed:
[mfem] Add a method to resolve all header dependencies of a spec using its
       link dependencies. Use the new method for hipblas.

[petsc] Add new variant, fortran-bindings, which is propagated to the
        petsc configuration via the option with the same name.

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.8

v-dobrev committed:
[mfem] Handle the libpmi dependency of cray-mpich

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.8

v-dobrev committed:
In Gitlab CI, since gcc 7 crashes when building mfem, try to use newer gcc
for building mfem

v-dobrev committed:
Fix typo

v-dobrev committed:
[mfem] Another try to use gcc@12 for mfem in some Gitlab CI pipelines

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.8

v-dobrev commented:
Migrated to https://github.com/spack/spack-packages/pull/496.
